packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts

```diff
@@ -156,8 +175,10 @@ function filterMutationRecords(
                 }
             } else {
                 // Mutation happened under a root node(shadow root or document) and the decision is straighforward
-                // Ascend the tree starting from target and check if observer is qualified
-                if (isQualifiedObserver(observer, target)) {
+                // First, check if the target is not the descendant of a lwc host element.
+                // If it is not, return record as-is.
+                // If it is, then ascend the tree starting from target and check if observer is qualified
+                if (isTrue(isNonShadowedElement(target)) || isQualifiedObserver(observer, target)) {
```

Comment:
usually for methods that are called `is*` we don't do the `isTrue()` around them. cc @ekashida